packages/logging/src/log.js

```diff
@@ -136,6 +137,27 @@ Log.formatName_ = function(projectId, name) {
   return path + name;
 };
 
+Log.prototype.assignDefaultMetadata_ = function(entry, callback) {
+  var entryMetadata = entry.metadata;
```

Comment:
I believe this function will currently modify the object the user is still holding. We should leave theirs untouched, but decorate it internally.